### DISCUSSION TITLE
Script help

### DISCUSSION THREAD

**Discussion by neilanthonystuckey** [rdswift/picard-scripts#3] [General]
Hello thanks for the awesome script. I use it quite a bit.  I was wondering if you could help me update the copied script to match my programs naming.  I can't adjust the program to match your script naming convention.  I have been trying to change your script to match the program to no avail.  I essentially need the picard naming script to match this.

~ {Artist NameFirstCharacter} ~/{Artist NameThe}
 ~ A ~/Artist Name, The

{[Release Year]} {Album Title}/{medium:0}-{track:00} {Track CleanTitle}
[2023] The Album Title/1-03 Track Title 1

**Comment by rdswift** [comment]
I'm sorry, but I really don't understand what your asking.  What script are you referring to?  What do you mean by your "programs naming"?  Is the program something other than Picard?  Are you asking how to change my file naming script to output in a different format?

**Reply by neilanthonystuckey** [reply]
thanks for your time.  This is the script i was referring to,

https://github.com/rdswift/picard-scripts/blob/master/naming_scripts/Bob_Swift/Bob%20Swift%27s%20Naming%20Script.pts

Yes i use an app not MBP that handles a lot of things automatically as far as music goes. I would prefer to use this app with MBP and your script.  The problem is the app has limited naming schemes.  I can get pretty close to your script within its naming conventions but things are different and cause problems for me. I am asking for your help to change the naming in your script to match the apps naming.
this is the syntax within the app for the naming

syntax in app:  
~ {Artist NameFirstCharacter} ~/{Artist NameThe}/{[Release Year]} {Album Title}/{medium:0}-{track:00} {Track CleanTitle}

what the file path looks like in Ubuntu nautilus:
 /music/~ A ~/Artist Name, The/[2023] The Album Title/1-03 Track Title 1

as you can see there are some differences which causes albums to not show up in my app because the naming is different.  I have tried but cannot get your script to match. I am not too great at scripting.

Thanks

Neil

**Reply by rdswift** [reply]
Do I understand this correctly...  You want me to re-write my Picard file naming script into a different scripting language for use with a different program (that you haven't identified and I don't know the scripting language or specification)?  Or is it that you want me to re-write the Picard script such that it renames the files and directories to match some format that your other (unspecified) program can parse?

You are aware that my file naming script outputs in a number of different formats depending on the artist, type of release and whether or not the music is classical, right?  If I were to write a custom script for you, I would need to know whether you still want the different formats or whether you want everything in one format.

**Reply by neilanthonystuckey** [reply]
What i am looking for is the change your scripts naming to output like this /music/~ A ~/Artist Name, The/[year] The Album Title/1-03 Track Title 1. I use your script and Music Brainz picard to cleanup tags and organize. I wish i can just do that in reverse and change the app to match your script but it has limited options for naming and the above is as close as i can get to your scripts naming. Was hoping it was fairly easy. I tried to edit your script in picard but i can get it to match like the example above.

**Reply by rdswift** [reply]
In that case, it appears that you don't need the full functionality of my file naming script.  Try changing your file naming script to the following and see if that does what you want.  You should also set the **Use standardized artist names** option to enabled (checked) in Picard's [Metadata Options](https://picard-docs.musicbrainz.org/en/config/options_metadata.html).

```
~ $firstalphachar(%albumartistsort%) ~/$swapprefix(%albumartist%)/[%year%] %album%/%discnumber%-$num(%tracknumber%,2) %track%
```
